### ISSUE TITLE
Improve error message for Get-Service with an invalid ComputerName

### ISSUE THREAD

**Issue by joeyaiello** [PowerShell/PowerShell#2747]
Filed on UserVoice: http://windowsserver.uservoice.com/forums/301869/suggestions/17193962

The error message for `Get-Service` returns a `NoServiceFoundForGivenName` exception even when the problem is an invalid `-ComputerName`, potentially misleading the user. 

Steps to reproduce
------------------
```powershell
Get-Service wuauserv -ComputerName MadeUpComputerName
``` 

Expected behavior
-----------------
```
Get-Service : Cannot find any computer with name 'MadeUpComputerName'.
At line:1 char:1
+ Get-Service wuauserv -ComputerName MadeUpComputerName
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (wuauserv:String) [Get-Service], ServiceCommandExce
   ption
    + FullyQualifiedErrorId : SomeOtherExceptionForComputerNameNotFound,Microsoft.PowerShell.Commands.GetService
   Command
```

Actual behavior
---------------
```
Get-Service : Cannot find any service with service name 'wuauserv'.
At line:1 char:1
+ Get-Service wuauserv -ComputerName MadeUpComputerName
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (wuauserv:String) [Get-Service], ServiceCommandExce
   ption
    + FullyQualifiedErrorId : NoServiceFoundForGivenName,Microsoft.PowerShell.Commands.GetService
   Command
```

Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSRemotingProtocolVersion      2.3
PSVersion                      6.0.0-alpha
WSManStackVersion              3.0
BuildVersion                   3.0.0.0
SerializationVersion           1.1.0.1
PSEdition                      Core
GitCommitId                    v6.0.0-alpha.12
CLRVersion
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
```

**Comment by SteveL-MSFT** [comment]
We should probably keep -computername for compat reasons, but maybe consider making it use PSRP under the covers

**Comment by joeyaiello** [comment]
@SteveL-MSFT maybe, but I think that's a much larger discussion (and body of work) than the issue here.

**Comment by kvprasoon** [comment]
This could completely break the logic of a script in case of error handling

**Comment by iSazonov** [comment]
Yes,it can. But on the other hand, adding the new exception we improve the error handling.

**Comment by kvprasoon** [comment]
@iSazonov  An example use case is - If there is something to do based on the exception returned , end user/script will be misguided and won't be able to do what it is intended to do.

**Comment by iSazonov** [comment]
I searched `SomeOtherExceptionForComputerNameNotFound` and `SomeOtherExceptionForComputerNameNotFound,Microsoft.PowerShell.Commands.GetService` on GitHub found nothing. Google return 69 (!) results.
Therefore we could assume that this _breaking_ change is not disruptive.
New major version allows breaking changes and any script should be tested with the new version, especially if we migrate from FullCLR to CoreCLR.

**Comment by Vivek2015** [comment]
You can handle the error like below, you can customize the same according to you needs
`
try
{
  Get-Service wuauserv -ComputerName MadeUpComputerName -ErrorAction Stop
}

catch [Microsoft.PowerShell.Commands.ServiceCommandException]
{
  # get error record
  [Management.Automation.ErrorRecord]$e = $_

  # retrieve information about runtime error
  $info = [PSCustomObject]@{
    Exception = $e.Exception.Message
    Reason    = $e.CategoryInfo.Reason
    Target    = $e.CategoryInfo.TargetName
    Script    = $e.InvocationInfo.ScriptName
    Line      = $e.InvocationInfo.ScriptLineNumber
    Column    = $e.InvocationInfo.OffsetInLine
  }
  
  # output information. Post-process collected info, and log info (optional)
  $info
}


`

**Comment by kvprasoon** [comment]
@Vivek2015  Problem here is not how to handle error , but giving right exception.

**Comment by KevinMarquette** [comment]
What if we used the same exception that `Test-Connection` gives?

    Testing connection to computer 'BAD_NAME' failed: No such host is known
    FullyQualifiedErrorId : TestConnectionException,Microsoft.PowerShell.Commands.TestConnectionCommand

**Comment by iSazonov** [comment]
@KevinMarquette Feel free to make the PR. In code review time it can be changed.
I would prefer something like `GetServiceRemoteConnectionException`

[Code point for fix](https://github.com/PowerShell/PowerShell/blob/02b5f357a20e6dee9f8e60e3adb9025be3c94490/src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs#L462).

It seems we should also fix the exception when Get-Service is called without service name.
```powershell
PS C:\Program Files\PowerShell\6.0.0.16> get-service -ComputerName sadsad
get-service : Cannot open Service Control Manager on computer 'sadsad'. This operation might require other privileges.
At line:1 char:1
+ get-service -ComputerName sadsad
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-Service], InvalidOperationException
    + FullyQualifiedErrorId : System.InvalidOperationException,Microsoft.PowerShell.Commands.GetServiceCommand

PS C:\Program Files\PowerShell\6.0.0.16> get-service wuauserv -ComputerName sadsad
get-service : Cannot find any service with service name 'wuauserv'.
At line:1 char:1
+ get-service wuauserv -ComputerName sadsad
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (wuauserv:String) [Get-Service], ServiceCommandException
    + FullyQualifiedErrorId : NoServiceFoundForGivenName,Microsoft.PowerShell.Commands.GetServiceCommand
```

**Comment by joandrsn** [comment]
This issue should probably be closed since issue #5090 which has been addressed in pull-request #5094

**Comment by iSazonov** [comment]
We removed `ComputerName` parameter at all in #5094
